products/bridge/smart-contracts/foundry/test/Relayer/Helpers.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.20;
+
+import "contracts/ValidatorManager.sol";
+import "contracts/Relayer.sol";
+import "foundry/test/Tester.sol";
+
+using ECDSA for bytes32;
+
+contract TransferReentrancyTester {
+    address target;
+    bytes data;
+    bool public alreadyEntered = false;
+
+    function testVulnerability(
+        address _target,
+        bytes calldata _data
+    ) external returns (bool) {
+        target = _target;
+        data = _data;
+
+        (bool success, ) = target.call(data);
+        return success;
+    }
+
+    receive() external payable {
+        if (address(target).balance > 0) {
```

Comment:
I'm wondering if this is the right condition for continuing the reentrancy attack. I would have expected something like `gasleft() > minGas` where `minGas` is the amount needed for one iteration, i.e. we stop if the remaining gas does not allow to reenter once more...

Reply:
yeah, this test i've realised is not very realistic. Transfer calls have a fixed limit of 2300 gas stipend. So it is actually not enough gas to ever successfully do a reentrancy.